Prototype of minhash

sukumar-poojari opened the pull request:
/*
This code takes the number of chunks to be stored in the segment and add the specified number of chunks into the segment.
Then it generates array of 20 minhashes of size 4 byte which are having the lowest digest values and store them in leveldb as key value pair
*/

Signed-off-by: Sukumar sukumarpoojari92@gmail.com

sukumar-poojari committed:
Prototype of minhash
/*
This code takes the number of chunks to be stored in the segment and add the specified number of chunks into the segment.
Then it generates array of 20 minhashes of size 4 byte which are having the lowest digest values and store them in leveldb as key value pair
*/

Signed-off-by: Sukumar <sukumarpoojari92@gmail.com>

josephaug26 commented:
why dont u use the same rabin_karp, vector, sha1 etc code from existing code ? why have another copy ?

sukumar-poojari committed:
Generation of minhash and finding similarity #26
/*The minhash generation can be done using two methods
1. XORing the hash value with range of prime numbers
2. By dividing the hash into 4 parts and finding the smallest minhashes out of that
Integrated this approach with yadl
The XOR part need to be corrected and deduping when there is a percentage of similarity above the threshold value*/

Signed-off-by: Sukumar <sukumarpoojari92@gmail.com>

sukumar-poojari committed:
Minhash Dedup #26:
/*
This patch contains the minhash way of deduplication.
The options are provided to dedupe using default way of minhash,
by using only one hash value or using the XOR way of minhash, where we find the minhash by
XORing a hash value with given range of prime numbers.
Based on the percentage of similarity found between the segments, they are deduped.
Inner deduplication of the segment is also done along with the creation of segment.
If the percentage of similarity is above the threshold value, then only the extra contents are appended to the matched segment,
and the matched contents will refer to the matched segment.
The files are restored using the minhash stub.
*/

Signed-off-by: Sukumar <sukumarpoojari92@gmail.com>

sukumar-poojari committed:
Minhash Dedup #26
/*
This patch contains the minhash way of deduplication.
The options are provided to dedupe using default way of minhash,
by using only one hash value or using the XOR way of minhash, where we find the minhash by
XORing a hash value with given range of prime numbers.
Based on the percentage of similarity found between the segments, they are deduped.
Inner deduplication of the segment is also done along with the creation of segment.
If the percentage of similarity is above the threshold value, then only the extra contents are appended to the matched segment,
and the matched contents will refer to the matched segment.
The files are restored using the minhash stub.
*/

Signed-off-by: Sukumar <sukumarpoojari92@gmail.com>
